src/backend/base/langflow/services/database/service.py

```diff
@@ -145,28 +146,40 @@ async def with_async_session(self):
         async with AsyncSession(self.async_engine, expire_on_commit=False) as session:
             yield session
 
-    async def migrate_flows_if_auto_login(self) -> None:
-        # if auto_login is enabled, we need to migrate the flows
-        # to the default superuser if they don't have a user id
-        # associated with them
+    async def assign_orphaned_flows_to_superuser(self) -> None:
+        """Assign flows without a user ID to the default superuser if auto_login is enabled."""
+        # Get the settings service to check if auto_login is enabled
         settings_service = get_settings_service()
         if settings_service.auth_settings.AUTO_LOGIN:
             async with self.with_async_session() as session:
-                # `== None` translates to SQL `is NULL`
+                # Select flows where user_id is NULL (orphaned flows)
                 stmt = select(models.Flow).where(models.Flow.user_id == None)  # noqa: E711
-                flows = (await session.exec(stmt)).all()
-                if flows:
-                    logger.debug("Migrating flows to default superuser")
-                    username = settings_service.auth_settings.SUPERUSER
-                    user = await get_user_by_username(session, username)
-                    if not user:
+                orphaned_flows = (await session.exec(stmt)).all()
+
+                if orphaned_flows:
+                    logger.debug("Assigning orphaned flows to the default superuser")
+
+                    # Get the default superuser
+                    superuser_username = settings_service.auth_settings.SUPERUSER
+                    superuser = await get_user_by_username(session, superuser_username)
+
+                    if not superuser:
                         logger.error("Default superuser not found")
-                        msg = "Default superuser not found"
-                        raise RuntimeError(msg)
-                    for flow in flows:
-                        flow.user_id = user.id
+                        raise RuntimeError("Default superuser not found")
+
+                    # Assign each orphaned flow to the superuser
+                    for flow in orphaned_flows:
+                        flow.user_id = superuser.id
+                        name_match = re.search(r"\((\d+)\)$", flow.name)
+                        if not name_match:
+                            flow.name = f"{flow.name} (1)"
+                        else:
+                            num = int(name_match.group(1)) + 1
+                            flow.name = re.sub(r"\(\d+\)$", f"({num})", flow.name)
+
+                    # Commit the changes to the database
```

Comment:
What do you think of checking if there are flows with this name before changing them? 

I'm saying this because adding a (1) to a flow that has no duplicates feels weird.